symengine/ruby/ext/symengine/symengine.c

```diff
@@ -22,4 +24,11 @@ void Init_symengine() {
     rb_define_method(c_basic, "/", cbasic_div, 1);
     rb_define_method(c_basic, "**", cbasic_pow, 1);
     rb_define_method(c_basic, "-@", cbasic_neg, 0);
+    rb_define_method(c_basic, "to_s", cbasic_to_str, 0);
+    rb_define_method(c_basic, "expand", cbasic_expand, 0);
+
+    //Symbol class
+    c_symbol = rb_define_class_under(m_symengine, "Symbol", c_basic);
+    rb_define_alloc_func(c_symbol, csymbol_alloc);
```

Comment:
That should be how it happens. Still I can't say for sure, because Ruby variables don't have a type, the data has the type and every Ruby object in the C API is a `VALUE` which is void pointer to the struct. All `VALUE`s have a `Type` which tells us the type of the data. There's a possibility that the `Basic` object has a pointer to the actual struct(that we allocated) that contains the information which it passes around when we want to call a function, and that function will work since it gets the pointer to the data that the `Basic` object wraps, the struct it was defined for, the actual struct that contains the information.